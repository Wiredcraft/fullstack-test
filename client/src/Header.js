import React, { useState, useEffect } from 'react';
import { Link } from '@reach/router';
import logo from './logo.png';
import LogInOrOut from './LogInOrOut';

const Header = ({ children }) => {

  const [ top, setTop ] = useState(document.body.scrollTop);

  const calcScrollTop = () => {
    setTop(document.body.scrollTop || document.documentElement.scrollTop);
    // console.log(top)
  };

  useEffect(() => {
    window.addEventListener('scroll', calcScrollTop);

    return function cleanup() {
      window.removeEventListener('scroll', calcScrollTop);
    };
  }, []);

  return (
    <header className={`${top > 100 ? 'minimized' : ''}`}>
      <div className='header-content body-container'>
        <Link to={`${process.env.PUBLIC_URL}/`} className='title-and-logo'>
          <img className='logo' src={logo} alt='' />
          <h1>Talk Lightning</h1>
        </Link>
        <LogInOrOut />
      </div>
    </header>
  );
}

export default Header;
