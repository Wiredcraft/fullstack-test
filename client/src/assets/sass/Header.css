@import url("https://fonts.googleapis.com/css?family=Roboto+Condensed:300,300i,400");
.c-Header {
  background-color: #FF5D73;
  padding: 10px;
  display: flex;
  line-height: 1.5;
  justify-content: space-between;
  color: #ffffff; }
  .c-Header__title {
    margin: 0;
    font-size: 1.3rem; }
  .c-Header__add-btn {
    background-color: transparent;
    font-size: 1rem;
    font-family: "Roboto Condensed", sans-serif; }
    .c-Header__add-btn:focus {
      outline: 0;
      background-color: #ff2a47; }
