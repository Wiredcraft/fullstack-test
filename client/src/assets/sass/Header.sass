@import './_variables.sass';

.c-Header
  background-color: $primary-color
  padding: 10px
  display: flex
  line-height: 1.5
  justify-content: space-between
  &__title
    margin: 0
    font-size: 1.3rem
  &__add-btn
    background-color: transparent
    font-size: 1rem
    border: 1px solid $text-color
    font-family: "Roboto Condensed", sans-serif
    &:focus
      outline: 0
      background-color: darken($primary-color, 10%)
