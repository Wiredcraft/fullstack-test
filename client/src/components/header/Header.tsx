import * as React from "react";

import { useAuth } from "../../contexts/AuthContext";
function Header() {
	const { isAuthenticated } = useAuth() // to decide whether is going to show login button or create talk button
	const [showLoginForm, setLoginState] = React.useState(false)
	const [showCreateTalkForm, setTalkState] = React.useState(false)
	
	return (
		<div>
			{
				if(!isAuthenticated){
					return <button >Open Login Form By Changing state</button>
				}
			}

			{
				if(isAuthenticated){
					return <button >Open Create talk form by Changing state</button>
				}
			}

			{
				if(showCreateTalkForm) {
					return <LoginForm />
				}
				if(showCreateTalkForm) {
					return <CreateTalkForm />
				}
			}

		</div>
	)
}