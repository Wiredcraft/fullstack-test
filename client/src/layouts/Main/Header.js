import React from 'react';
import { Auth } from 'Components';
import { Link, withRouter } from 'react-router-dom';
import styles from './style.module.less';

class Header extends React.Component {
    static contextType = Auth.Context;

    onLogout = () => {
        app.storage.remove('user');
        app.storage.remove('authToken');
        this.context.notifyUserChange();
        this.props.history.push("/");
    }

    render() {
        const { user } = this.context;
        return (
            <div className={styles.header}>
                <div className={styles.inner}>
                    <Link to={{ pathname: "/" }} className={styles.logo}>
                        <img src={require('Assets/logo.png').default} />
                    </Link>
                    <h1 className={styles.title}>Lighting Talk</h1>
                    {!user && (
                        <>
                            <Link to={{ pathname: '/register' }}>register</Link>
                            <span className={styles.sep}>|</span>
                            <Link to={{ pathname: '/login' }}>login</Link>
                        </>
                    )}
                    {user && (
                        <>
                            <Link to={{ pathname: '/addTalk'}}>add Talk</Link>
                            <span className={styles.sep}>|</span>
                            <Link to={{ pathname: '/' }}>{user.name}</Link>
                            <span className={styles.sep}>|</span>
                            <Link onClick={this.onLogout}>logout</Link>
                        </>
                    )}
                </div>
            </div>
        )
    }
}

export default withRouter(Header);