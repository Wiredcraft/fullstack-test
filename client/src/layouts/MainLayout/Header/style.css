.header-box {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;

  height: 100%;
  background: var(--primary-color);
  color: #ffffff;
}

.header-logo {
  display: block;

  width: var(--layout-header-height);
  text-align: center;
}

.header-title {
  margin: 0;

  font-size: 26px;
  font-weight: 900;
}

.header-login {
  margin: 0 8px 0 auto;
  color: inherit;
}
