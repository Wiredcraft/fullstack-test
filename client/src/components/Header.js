import React from 'react';
import '../assets/sass/Header.css'

const Header = (props) =>
  <div className='c-Header'> 
    <h1 className='c-Header__title'>Hacker Talks</h1>
    <button
      className='c-btn'
      onClick={props.onAddClick}
    >
      &#10133; add
    </button>
  </div>

export default Header;
