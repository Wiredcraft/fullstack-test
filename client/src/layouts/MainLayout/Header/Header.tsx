import { useRecoilValue } from 'recoil';
import { useNavigate } from 'react-router-dom';

import style from './style.css';

import { Logo } from '@/components/Logo';
import { myAtom } from '@/store/auth';
import { createStyler } from '@/utils/styler';
import { Button } from '@/components/Button';
import { PATH } from '@/const';

const styler = createStyler(style);

interface Props {
  className?: string;
}

export function Header(props: Props) {
  const { className } = props;

  const user = useRecoilValue(myAtom);

  const navigate = useNavigate();
  const handleLoginClick = () => {
    navigate(PATH.LOGIN);
  };

  return (
    <div className={styler(className, 'header-box')}>
      <div className={styler('header-logo')}>
        <Logo>W</Logo>
      </div>

      <h1 className={styler('header-title')}>Lightning Talks</h1>

      {user == null ? (
        <Button className={styler('header-login')} onClick={handleLoginClick}>
          Login
        </Button>
      ) : (
        <></>
      )}
    </div>
  );
}
