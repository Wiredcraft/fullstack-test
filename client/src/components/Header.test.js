import React from 'react';
import { render } from 'react-testing-library';
import Header from './Header';

it('renders Header correctly', () => {
  const { asFragment } = render(<Header />);

  expect(asFragment()).toMatchSnapshot();
});
