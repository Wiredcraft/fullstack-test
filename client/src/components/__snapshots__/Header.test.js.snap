// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
<DocumentFragment>
  <header>
    <div
      class="header-content body-container"
    >
      <a
        aria-current="page"
        class="title-and-logo"
        href="/"
      >
        <img
          alt=""
          class="logo"
          src="logo.png"
        />
        <h1>
          Talk Lightning
        </h1>
      </a>
      <div
        class="log-in-or-out"
      >
        <button
          class="styled-link"
          hidden=""
        >
          Logout
        </button>
        <div
          id="firebaseui-auth-container"
        />
      </div>
    </div>
  </header>
</DocumentFragment>
`;
